lang/py/avro/io.py

```diff
@@ -776,6 +766,9 @@ def read_enum(self, writers_schema: avro.schema.EnumSchema, readers_schema: avro
         # read data
         index_of_symbol = decoder.read_int()
         if index_of_symbol >= len(writers_schema.symbols):
+            default = writers_schema.default
```

Comment:
From my comment on AVRO-3760 -- if we're a bit more strict, this should probably fail here when there's not a readers_schema.  We should discuss on the JIRA what we actually want to do.